### PR TITLE
Bucket test updates

### PR DESCRIPTION
**Pull request by brunoabinader**
Another spin-off from #9148, this PR adds simple bucket feature insertion tests, and also:
- Check if bucket contains data in `Bucket::needsUpload()`
- Remove redundant `GeometryCollection` param in `Bucket::addFeature()`

### CODE REVIEW

**Comment by brunoabinader** [comment]
We can further expand `Buckets.FillBucket` to test edge cases when handling with Earcut internally, but so far I haven't found a way to provoke the same crash backlog as seen in #9148.

File: src/mbgl/tile/geometry_tile_worker.cpp [hunk 1]
```diff
@@ -323,9 +323,8 @@ void GeometryTileWorker::redoLayout() {
                 if (!filter(feature->getType(), feature->getID(), [&] (const auto& key) { return feature->getValue(key); }))
                     continue;
 
-                GeometryCollection geometries = feature->getGeometries();
```
Comment (jfirebaugh) [inline src/mbgl/tile/geometry_tile_worker.cpp]: Please revert this; calling `getGeometries()` only once per feature was an important performance optimization. (The result is constructed on the fly and not cached by `VectorTileFeature`.)

Comment (brunoabinader) [inline src/mbgl/tile/geometry_tile_worker.cpp]: Ah, makes sense - ok I'll just add this as a comment 👍

File: src/mbgl/renderer/bucket.hpp [hunk 2]
```diff
@@ -28,6 +28,9 @@ class Bucket : private util::noncopyable {
     Bucket() = default;
     virtual ~Bucket() = default;
 
+    // Feature geometries are also used to populate the feature index.
+    // Obtaining these is a costly operation, so we do it only once, and
+    // pass-by-const-ref the geometries as a second parameter.
```
Comment (jfirebaugh) [inline src/mbgl/renderer/bucket.hpp]: 👍
